### PR TITLE
chore: scroll to the top of the page when the search box value changes

### PR DESCRIPTION
**Pull request by crazyair**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at f44497d</samp>

Improved component overview page navigation by scrolling to sections on menu click. Added `scroll-into-view-if-needed` dependency to `.dumi/theme/builtins/ComponentOverview/index.tsx`.

### <samp>🤖 Generated by Copilot at f44497d</samp>

* Import `scroll-into-view-if-needed` package to enable smooth scrolling to component sections ([link](https://github.com/ant-design/ant-design/pull/45211/files?diff=unified&w=0#diff-f1df5cf9560d26c9862a4824e7d1a13f6a8bec1ec7338e9327f3e7a6a74c1585L7-R10))
* Call `scrollIntoView` function with `sectionRef` and options to scroll to the selected component section when clicking on the sidebar menu ([link](https://github.com/ant-design/ant-design/pull/45211/files?diff=unified&w=0#diff-f1df5cf9560d26c9862a4824e7d1a13f6a8bec1ec7338e9327f3e7a6a74c1585R161-R170))

### CODE REVIEW

File: .dumi/theme/builtins/ComponentOverview/index.tsx [hunk 1]
```diff
@@ -157,6 +158,16 @@ const Overview: React.FC = () => {
             onChange={(e) => {
               setSearch(e.target.value);
               reportSearch(e.target.value);
+              if (sectionRef.current) {
+                scrollIntoView(sectionRef.current, {
+                  scrollMode: 'if-needed',
+                  block: 'start',
+                  behavior: (actions) =>
+                    actions.forEach(({ el, top }) => {
+                      el.scrollTop = top - 64;
```
Comment (afc163) [inline .dumi/theme/builtins/ComponentOverview/index.tsx:167]: 这个 64 最好不要写死，有地方能拿到么

**Comment by afc163** [comment]
写一下修什么问题。

File: .dumi/theme/builtins/ComponentOverview/index.tsx (hunk 1 above)
Comment (crazyair) [reply .dumi/theme/builtins/ComponentOverview/index.tsx:167]: 64 是 顶部导航栏高度，应该不会经常变，需要动态的获取吗

**Comment by crazyair** [comment]
> 写一下修什么问题。

在滚动页面后，再搜索，搜索的结果看不到


https://github.com/ant-design/ant-design/assets/7971419/11a34436-5cf0-4670-8a8d-7b5c8e3ef4de

File: .dumi/theme/builtins/ComponentOverview/index.tsx (hunk 1 above)
Comment (afc163) [reply .dumi/theme/builtins/ComponentOverview/index.tsx:167]: 问题不大。

**Review by afc163 — APPROVED** [review]

**Review by afc163 — APPROVED** [review]

**Pull request merged**